posthog/api/capture.py

```diff
@@ -268,7 +269,12 @@ def drop_events_over_quota(token: str, events: List[Any]) -> List[Any]:
             if token in limited_tokens_recordings:
                 EVENTS_DROPPED_OVER_QUOTA_COUNTER.labels(resource_type="recordings", token=token).inc()
                 if settings.QUOTA_LIMITING_ENABLED:
-                    continue
+                    # tell clients to wait until midnight before trying again, or at least 1 hour, if midnight is soon
+                    seconds_until_midnight = max(
+                        (datetime.now() + timedelta(days=1)).replace(hour=0, minute=0, second=0) - datetime.now(),
+                        timedelta(hours=1),
+                    ).total_seconds()
+                    raise Throttled(detail="Recording event dropped due to billing limit", wait=seconds_until_midnight)
```

Comment:
Personally I would simplify this greatly. Their usage could be reset either by the hourly check, or the midnight check, or the owner changing the billing limits. As such I would lean towards a **simple 10 minute backoff.** 
That way the maximum negative impact would be ~10 minutes after the quota throttle is lifted.